src/tcp_connections.rs

```diff
@@ -206,6 +206,71 @@ mod test {
       assert_eq!(10, responses.len());
     }
 
+#[test]
+    fn test_multiple_client_small_stream() {
+        const MSG_COUNT: usize = 5;
+        const CLIENT_COUNT: usize = 101;
+
+        let (listener, u32) = listen().unwrap();
+        let mut vector_senders = Vec::new();
+        let mut vector_receiver = Vec::new();
+        for _ in 0..CLIENT_COUNT {
+            let (i, o) = connect_tcp(SocketAddr::from_str("127.0.0.1:5483").unwrap()).unwrap();
+            let boxed_output: Box<OutTcpStream<u64>> = Box::new(o);
+            vector_senders.push(boxed_output);
+            let boxed_input: Box<InTcpStream<(u64, u64)>> = Box::new(i);
+            vector_receiver.push(boxed_input);
+        }
+
+        //  send
+        for mut v in &mut vector_senders {
+            for x in 0u64 .. MSG_COUNT as u64 {
+                if v.send(&x).is_err() { break; }
+            }
+        }
+
+        //  close sender
+        loop {
+           let sender = match vector_senders.pop() {
+                None => break, // empty
+                Some(sender) => sender.close(),
+            };
+        }
+
+        // listener
+        thread::spawn(move || {
+            for (connection, u32) in listener.into_blocking_iter() {
+                // Spawn a new thread for each connection that we get.
+                thread::spawn(move || {
+                    let (i, mut o) = upgrade_tcp(connection).unwrap();
+                    for x in i.into_blocking_iter() {
+                        if o.send(&(x, x + 1)).is_err() { break; }
```

Comment:
This will not compile for now, may be a compiler bug, unsure, it needs to be 
```   let i = i.blocking_iter(); 
       let vec : Vec<u64> = i.collect();
        for &x in vec.iter() {
```